web/template.go

```diff
@@ -0,0 +1,50 @@
+package web
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"text/template"
+
+	"fmt"
+
+	"github.com/feedhenry/negotiator/deploy"
+	"github.com/feedhenry/negotiator/pkg/openshift"
+)
+
+// TemplateHandler is the handler for the /services/template route
+type TemplateHandler struct {
+	tl deploy.TemplateLoader
+	td deploy.TemplateDecoder
+}
+
+func NewTemplateHandler(tl deploy.TemplateLoader, td deploy.TemplateDecoder) TemplateHandler {
+	return TemplateHandler{
+		tl: tl, td: td,
+	}
+}
+
+// List all available templates
+func (s TemplateHandler) List(res http.ResponseWriter, req *http.Request) {
+	res.Header().Add("Content-type", "application/json")
+	var err error
+	var t *template.Template
+	var ts []*deploy.Template
+	for v := range openshift.PackagedTemplates {
+		t, err = s.tl.Load(v)
+		if err != nil {
+			fmt.Println(err)
+		}
+		var buf bytes.Buffer
+		if err := t.ExecuteTemplate(&buf, v, &deploy.Payload{}); err != nil {
+			fmt.Println(err)
+		}
+
+		decoded, err := s.td.Decode(buf.Bytes())
+		if err != nil {
+			fmt.Println(err)
+		}
+		ts = append(ts, decoded)
```

Comment:
I might have misread this code, but it looks like this is returning the entire template in the JSON response?
Would we not want the response to be something like:
```json
{
    "services": ["cache", "mongodb"]
}
```